operate/services/manage.py

```diff
@@ -551,23 +551,34 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
             service.store()
         self.logger.info(f"Service state: {chain_data.on_chain_state.name}")
 
-        if user_params.use_staking:
-            staking_params = sftxb.get_staking_params(
+        current_staking_program = self._get_current_staking_program(service, chain)
+        fallback_params = dict(  # nosec
+            staking_contract=NULL_ADDRESS,
+            agent_ids=[user_params.agent_id],
+            service_registry="0x9338b5153AE39BB89f50468E608eD9d764B755fD",  # nosec
+            staking_token=NULL_ADDRESS,  # nosec
+            service_registry_token_utility="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8",  # nosec
+            min_staking_deposit=20000000000000000000,
+            activity_checker=NULL_ADDRESS,  # nosec
+        )
+
+        current_staking_params = sftxb.get_staking_params(
+            fallback_params=fallback_params,
+            staking_contract=get_staking_contract(
+                chain=ledger_config.chain,
+                staking_program_id=current_staking_program,
+            ),
+        )
+        if user_params.staking_program_id == current_staking_program:
+            target_staking_params = current_staking_program
+        else:
```

Comment:
Why this `if`?  If it's an optimization to avoid calling staking methods, the `get_staking_params` method already has an internal cache for staking parameters.

Reply:
oh yeah, I did not see it